libraries/chain/db_market.cpp

```diff
@@ -229,8 +303,14 @@ bool maybe_cull_small_order( database& db, const limit_order_object& order )
     */
    if( order.amount_to_receive().amount == 0 )
    {
-      //ilog( "applied epsilon logic" );
-      db.cancel_order(order);
+      if( order.deferred_fee > 0 && db.head_block_time() <= HARDFORK_CORE_604_TIME )
+      {
+         wlog( "At block ${n}, cancelling order without charging a fee: ${o}", ("n",db.head_block_num())("o",order) );
+         db.cancel_limit_order( order, true, true );
+      }
+      else
+         db.cancel_limit_order( order );
+      // TODO check call orders here?
```

Comment:
Please resolve this TODO.

Reply:
When #606 is implemented (done some work in [this commit](https://github.com/bitshares/bitshares-core/commit/bba18da134975e8bd5624b2dd77fb070923344a7)), will be no need to check call orders after a limit order is cancelled.